kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -103,6 +148,22 @@ class KinematicsBase
                              const kinematics::KinematicsQueryOptions &options = kinematics::KinematicsQueryOptions()) const = 0;
 
   /**
+   * @brief Given a desired pose of the end-effector, compute the set joint angles solutions that are able to reach it.  This
+   * is a default implementation that returns only one solution and so its result is equivalent to calling 'getPositionIK(...)'
+   * with a zero initialized seed.
+   * @param ik_pose the desired pose of the tip frame
+   * @param solutions a vector of vectors where each entry is a valid solution
+   * @param result a struct that reports the results of the query
+   * @param an option struct which contains the type of redundancy discretization used.  This default implementation only
+   * supports the KinmaticSearches::ONE discretization; requesting any other will result in failure.
+   * @return True if a valid set of solutions was found, false otherwise
+   */
+  virtual bool getMultipleIK(const geometry_msgs::Pose &ik_pose,
+                             std::vector< std::vector<double> >& solutions,
+                             KinematicsResult& result,
+                             const kinematics::KinematicsQueryOptions &options) const;
+
```

Comment:
I can understand this function for 6 DOF robots - gives you elbow down, elbow up solutions, etc. What is this function expected to do for redundant robots - the notion of "all" solutions is ill-defined here since technically there are infinite solutions.